warp10/src/main/java/io/warp10/continuum/egress/EgressExecHandler.java

```diff
@@ -285,6 +303,66 @@ public void handle(String target, Request baseRequest, HttpServletRequest req, H
 
       boolean terminate = false;
 
+      //
+      // Read the X-Warp10-Timebox header
+      //
+
+      String timebox = req.getHeader(Constants.HTTP_HEADER_TIMEBOX);
+      long maxtime = 0L;
+
+      if (null != timebox) {
+        boolean fromToken = false;
+        try {
+          maxtime = Long.parseLong(timebox) * Constants.TIME_UNITS_PER_MS;
+        } catch (NumberFormatException nfe) {
+          try {
+            maxtime = DURATION.parseDuration(new Instant(), timebox, false, false);
+          } catch (Exception ee) {
+            resp.sendError(errorCode, "Invalid value for header " + Constants.HTTP_HEADER_TIMEBOX + " " + ThrowableUtils.getErrorMessage(ee));
+            return;
+          }
+        }
+
+        //
+        // Check the configured maxtime as the upper limit unless a valid token was specified as timebox
+        //
+
+        if (!fromToken && 0 != MAXTIME && maxtime > 0) {
+          maxtime = Math.min(maxtime, MAXTIME);
+        }
+      } else {
+        maxtime = MAXTIME;
+      }
+
+      long timeLimit = MAXTIME;
+
+      if (null != TIMEBOX.TIMEBOX_MAXTIME_CAPNAME && null != Capabilities.get(stack, TIMEBOX.TIMEBOX_MAXTIME_CAPNAME)) {
+        String val = Capabilities.get(stack, TIMEBOX.TIMEBOX_MAXTIME_CAPNAME).trim();
+
+        if (val.startsWith("P")) {
+          timeLimit = DURATION.parseDuration(new Instant(), val, false, false);
+        } else {
+          try {
+            timeLimit = Long.valueOf(Capabilities.get(stack, TIMEBOX.TIMEBOX_MAXTIME_CAPNAME)) * Constants.TIME_UNITS_PER_MS;
+          } catch (NumberFormatException nfe) {
+          }
+        }
+      }
+
+      if (maxtime > timeLimit) {
+        maxtime = timeLimit;
+      }
```

Comment:
I thought the token capability should increase the time limit, not reduce it.

I think the logic should be:
```
long maxtime = MAXTIME;
long maxtimeHeader = // parse from header
long maxtimeCapability = // parse from capabilties

if (maxtimeCapability > 0) {
  maxtime = Math.max(maxtime, maxtimeCapability);
} else {
  maxtime = 0;
}

if (maxtimeHeader > 0) {
  maxtime = Math.min(maxtime, maxtimeHeader);
}
```